rmgpy/molecule/group.py

```diff
@@ -1262,11 +1273,42 @@ def specifyAtomExtensions(self,i,basename,R):
             if len(old_atom_type ) > 1:
                 old_atom_type_str = ''
                 for k in old_atom_type:
-                    old_atom_type_str += k.label
+                    labelList.append(k.label)
+                for p in sorted(labelList):
+                    old_atom_type_str += p
             else:
                 old_atom_type_str = old_atom_type[0].label
 
-            grps.append((grp,grpc,basename+'_'+old_atom_type_str+'->'+item.label,'atomExt',(i,)))
+            grps.append((grp,grpc,basename+'_'+str(i+1)+old_atom_type_str+'->'+item.label,'atomExt',(i,)))
+            
+        return grps
+    
+    def specifyRingExtensions(self,i,basename):
+        """
+        generates extensions for specifying if an atom is in a ring
+        """
+        cython.declare(grps=list,labelList=list,grp=Group,grpc=Group,atom_type=list,atom_type_str=str,k=AtomType,p=str)
+        
+        grps = []
+        labelList = []
+        
+        grp = deepcopy(self)
+        grpc = deepcopy(self)
+        grp.atoms[i].props['inRing'] = True
+        grpc.atoms[i].props['inRing'] = False
+        
+        atom_type = grp.atoms[i].atomType
+        
+        if len(atom_type ) > 1:
+            atom_type_str = ''
+            for k in atom_type:
+                labelList.append(k.label)
+            for p in sorted(labelList):
+                atom_type_str += p
+        else:
+            atom_type_str = atom_type[0].label
+        
+        grps.append((grp,grpc,basename+'_'+str(i+1)+atom_type_str+'-inRing','ringExt',(i,)))
```

Comment:
What about using `r0` or `r1` like in the adjlists instead of `-inRing`?

Reply:
Honestly, in practice the names are so long I've tried to err on the side of making it easy to read individual sections of the name (having a bit more detail) rather than the thing as a whole.  In particular `-inRing` is a less obvious extension so I thought I'd make it clearer in the naming, I'm not opposed to shortening it, but I feel like r0 and r1 is too short.